tests/multi_objective_tests/samplers_tests/test_motpe.py

```diff
@@ -63,545 +75,7 @@ def test_sample_independent_seed_fix() -> None:
-
-    assert len(set(suggestions)) == 1
-
-
-def test_get_observation_pairs() -> None:
-    def objective(trial: optuna.multi_objective.trial.Trial) -> Tuple[float, float]:
-        trial.suggest_int("x", 5, 5)
-        return 5.0, 5.0
-
-    sampler = MOTPEMultiObjectiveSampler(seed=0)
-    study = optuna.multi_objective.create_study(
-        directions=["minimize", "maximize"], sampler=sampler
-    )
-    study.optimize(objective, n_trials=5)
-
-    assert _motpe._get_observation_pairs(study, "x") == (
-        [5.0, 5.0, 5.0, 5.0, 5.0],
-        [[5.0, -5.0], [5.0, -5.0], [5.0, -5.0], [5.0, -5.0], [5.0, -5.0]],
-    )
-    assert _motpe._get_observation_pairs(study, "y") == (
-        [None, None, None, None, None],
-        [[5.0, -5.0], [5.0, -5.0], [5.0, -5.0], [5.0, -5.0], [5.0, -5.0]],
-    )
-
-
-def test_calculate_nondomination_rank() -> None:
-    # Single objective
-    test_case = np.asarray([[10], [20], [20], [30]])
-    ranks = list(_motpe._calculate_nondomination_rank(test_case))
-    assert ranks == [0, 1, 1, 2]
-
-    # Two objectives
-    test_case = np.asarray([[10, 30], [10, 10], [20, 20], [30, 10], [15, 15]])
-    ranks = list(_motpe._calculate_nondomination_rank(test_case))
-    assert ranks == [1, 0, 2, 1, 1]
-
-    # Three objectives
-    test_case = np.asarray([[5, 5, 4], [5, 5, 5], [9, 9, 0], [5, 7, 5], [0, 0, 9], [0, 9, 9]])
-    ranks = list(_motpe._calculate_nondomination_rank(test_case))
-    assert ranks == [0, 1, 0, 2, 0, 1]
-
-
-def test_solve_hssp() -> None:
-    sampler = MOTPEMultiObjectiveSampler(seed=0)
-
-    random.seed(128)
-
-    # Two dimensions
-    for i in range(8):
-        subset_size = int(random.random() * i) + 1
-        test_case = np.asarray([[random.random(), random.random()] for _ in range(8)])
-        r = 1.1 * np.max(test_case, axis=0)
-        truth = 0.0
-        for subset in itertools.permutations(test_case, subset_size):
-            truth = max(truth, sampler._compute_hypervolume(np.asarray(subset), r))
-        indices = sampler._solve_hssp(test_case, np.arange(len(test_case)), subset_size, r)
-        approx = sampler._compute_hypervolume(test_case[indices], r)
-        assert approx / truth > 0.6321  # 1 - 1/e
-
-    # Three dimensions
-    for i in range(8):
-        subset_size = int(random.random() * i) + 1
-        test_case = np.asarray(
-            [[random.random(), random.random(), random.random()] for _ in range(8)]
-        )
-        r = 1.1 * np.max(test_case, axis=0)
-        truth = 0
-        for subset in itertools.permutations(test_case, subset_size):
-            truth = max(truth, sampler._compute_hypervolume(np.asarray(subset), r))
-        indices = sampler._solve_hssp(test_case, np.arange(len(test_case)), subset_size, r)
-        approx = sampler._compute_hypervolume(test_case[indices], r)
-        assert approx / truth > 0.6321  # 1 - 1/e
+        _ = sampler.sample_independent(study, trial, "param-a", dist)
 
 
 def frozen_trial_factory(
```

Comment:
frozen_trial_factory is currently defined three times in the optuna directory. Can they be consolidated?

Reply:
Thanks for the proposal. It seems reasonable to me but should be done in another PR. What do you think?

Reply:
Agreed.